tests/test_cli_compile.py

```diff
@@ -3838,3 +3838,211 @@ def test_stdout_should_not_be_read_when_stdin_is_not_a_plain_file(
+)
+def test_second_order_requirements_relative_path_in_separate_dir(
+    pip_conf, runner, tmp_path, monkeypatch, relative_requirement_location
+):
+    """
+    As in the above test, nested requirements file path handling.
+    But in this case, the primary variable is the relative location of the two
+    requirements.
+
+    The expectation is that the output path will be relative to the current
+    working directory, *not* relative to the dir containing the initial
+    requirements file.
+
+    Parent dir layout:
+       - root_dir/
+         - requirements2.in
+         - subdir/
+           - requirements.in  # containing '-r ../requirements2.in'
+
+    Subdir layout:
+       - root_dir/
+         - requirements.in  # containing '-r ./subdir/requirements2.in'
+         - subdir/
+           - requirements2.in
+
+    Sibling dir layout:
+       - root_dir/
+         - subdir1/
+           - requirements.in  # containing '-r ../subdir2/requirements2.in'
+         - subdir2/
+           - requirements2.in
+    """
+    pip_current_version = get_pip_version_for_python_executable(sys.executable)
+    expect_path_normalization = pip_current_version >= Version("24.3")
+
+    # setup the two files in the requested dir layout
+    if relative_requirement_location == "parent":
+        (tmp_path / "subdir").mkdir()
+        req_in = tmp_path / "subdir" / "requirements.in"
+        req_in.write_text("-r ../requirements2.in\n")
+        req_in2 = tmp_path / "requirements2.in"
+        req_in2.write_text("small-fake-a\n")
+
+        if expect_path_normalization:
+            output_path = "requirements2.in"
+        else:
+            output_path = "subdir/../requirements2.in"
+    elif relative_requirement_location == "subdir":
+        req_in = tmp_path / "requirements.in"
+        req_in.write_text("-r ./subdir/requirements2.in\n")
+        (tmp_path / "subdir").mkdir()
+        req_in2 = tmp_path / "subdir" / "requirements2.in"
+        req_in2.write_text("small-fake-a\n")
+
+        if expect_path_normalization:
+            output_path = "subdir/requirements2.in"
+        else:
+            output_path = "./subdir/requirements2.in"
+    elif relative_requirement_location == "sibling_dir":
+        (tmp_path / "subdir1").mkdir()
+        req_in = tmp_path / "subdir1" / "requirements.in"
+        req_in.write_text("-r ../subdir2/requirements2.in\n")
+        (tmp_path / "subdir2").mkdir()
+        req_in2 = tmp_path / "subdir2" / "requirements2.in"
+        req_in2.write_text("small-fake-a\n")
+
+        if expect_path_normalization:
+            output_path = "subdir2/requirements2.in"
+        else:
+            output_path = "subdir1/../subdir2/requirements2.in"
+    else:
+        # test params didn't match body
+        raise NotImplementedError(relative_requirement_location)
+
+    # the input is given relative to the starting dir
+    input_path = req_in.relative_to(tmp_path).as_posix()
```

Comment:
I think the one you pass into CLI shouldn't matter. Only the paths that end up written to output would be important.